design/admin/templates/ajaxuploader/location.tpl

```diff
@@ -3,6 +3,20 @@
     <fieldset>
         <legend>{"Step 2/3: Choose a location for the new '%class' object"|i18n( 'design/admin/ajaxuploader', '', hash( '%class', $class.name|wash() ) )}</legend>
         <p>{'<em>%file</em> has successfully been uploaded.'|i18n( 'design/admin/ajaxupload', '', hash( '%file', $file.original_filename|wash() ) )}</p>
+        {if $class.identifier|eq( 'image' )}
+            {def $alt_text_required = ''}
+            {foreach $class.data_map as $attribute}
+                {if $attribute.data_type_string|eq( 'ezimage' )}
+                    {if $attribute.data_int2}
+                        {set $alt_text_required = ' (required)'}
+                    {/if}
+                {/if}
+            {/foreach}
+            <p>
+                <label for="alt_text_entry">Alternative text{$alt_text_required}:</label>
+                <input{if $alt_text_required} required class="input-required"{/if} id="UploadAlternativeText" name="UploadAlternativeText" size="70" maxlength="150">
+            </p>
+        {/if}
         <p>{"Please choose a location for the '%class' object that is going to be created from it."|i18n( 'design/admin/ajaxuploader', '', hash( '%class', $class.name|wash() ) )}</p>
```

Comment:
I've restructured the output here to support messaging in the main location.  I removed the inline conditional statements and instead I am storing the output in strings based on the upload.ini settings and and the status of the mandatory alt tag